drivers/bridge/bridge.go

```diff
@@ -673,40 +682,54 @@ func (d *driver) EndpointInfo(nid, eid types.UUID) (map[string]interface{}, erro
 }
 
 // Join method is invoked when a Sandbox is attached to an endpoint.
-func (d *driver) Join(nid, eid types.UUID, sboxKey string, options map[string]interface{}) (*driverapi.JoinInfo, error) {
+func (d *driver) Join(nid, eid types.UUID, sboxKey string, jinfo driverapi.JoinInfo, options map[string]interface{}) error {
 	network, err := d.getNetwork(nid)
 	if err != nil {
-		return nil, err
+		return err
 	}
 
-	if !network.config.EnableICC {
-		return nil, d.link(nid, eid, options, true)
+	endpoint, err := network.getEndpoint(eid)
+	if err != nil {
+		return err
 	}
 
-	return nil, nil
-}
+	if endpoint == nil {
+		return EndpointNotFoundError(eid)
+	}
 
-// Leave method is invoked when a Sandbox detaches from an endpoint.
-func (d *driver) Leave(nid, eid types.UUID, options map[string]interface{}) error {
-	network, err := d.getNetwork(nid)
+	for _, iNames := range jinfo.InterfaceNames() {
+		if iNames.ID() == 1 {
```

Comment:
Can you please explain what use of interface ID are you doing here ?
It gets set in CreateEndpoint() and then .. ? How could the interface not have ID == 1?
I think adding comment on what you are asserting here would help.